scalalib/src/mill/scalalib/spotless/SpotlessModule.scala

```diff
@@ -0,0 +1,87 @@
+package mill.scalalib.spotless
+
+import mill._
+import mill.define.{Command, Discover, ExternalModule, Sources, Task}
+import mill.api.{PathRef, Result}
+import mill.scalalib._
+import mill.main.Tasks
+import mainargs.arg
+
+trait SpotlessModule extends JavaModule {
+
+  def reformat(): Command[Unit] = T.command {
+    SpotlessWorkerModule
+      .worker()
+      .reformat(
+        filesToFormat(sources()),
+        resolvedSpotlessConfig()
+      )
+  }
+
+  def checkFormat(): Command[Unit] = T.command {
+    SpotlessWorkerModule
+      .worker()
+      .checkFormat(
+        filesToFormat(sources()),
+        resolvedSpotlessConfig()
+      )
+  }
+
+  def spotlessConfig: T[Seq[PathRef]] = T.sources(
+    T.workspace / ".spotless.conf",
+    os.pwd / ".spotless.conf"
+  )
+
+  private[SpotlessModule] def resolvedSpotlessConfig: Task[PathRef] = T.task {
+    val locs = spotlessConfig()
+    locs.find(p => os.exists(p.path)) match {
+      case None => Result.Failure(
+        s"None of the specified `spotlessConfig` locations exist. Searched in: ${locs.map(_.path).mkString(", ")}"
+      )
+      case Some(c) => Result.Success(c)
+    }
+  }
```

Comment:
Looks like we have
* a lookup path for config files
* a logic to find the best config file
* a final config to be used
* and some error handling in case there is no config file found

Lets change to user-facing tasks to enable users to explicitly specify a config

```scala
def spotlessConfig: T[PathRef]
```

It can be derived from a 

```scala
def spotlessConfigLookupPaths: T[Seq[PathRef]
```

task. I'm not sure, whether we really need it though. It could also be an implementation detail in `spotlessConfig` default impl.